src/server/game/Spells/Spell.cpp

```diff
@@ -1283,6 +1283,180 @@ void Spell::SelectImplicitCasterDestTargets(SpellEffIndex effIndex, SpellImplici
             dest = SpellDestination(x, y, liquidLevel, m_caster->GetOrientation());
             break;
         }
+        case TARGET_DEST_CASTER_FRONT_LEAP:
+        {
+            float distance = m_spellInfo->Effects[effIndex].CalcRadius(m_caster);
+            Map* map = m_caster->GetMap();
+            uint32 mapid = m_caster->GetMapId();
+            uint32 phasemask = m_caster->GetPhaseMask();
+            float x, y, z;
+            float destx, desty, destz, ground;
+            float orientation = m_caster->GetOrientation();
+
+            m_caster->GetPosition(x, y, z);
+            destx = x + distance * cos(orientation);
+            desty = y + distance * sin(orientation);
+
+            ground = map->GetHeight(phasemask, x, y, z);
+
+            if (!m_caster->HasUnitMovementFlag(MOVEMENTFLAG_FALLING) || (z - ground < 25.0f))
+            {
+                float tstX, tstY, tstZ, prevX, prevY, prevZ;
```

Comment:
C++ doesn't support variable hoisting like JavaScript does, that's why we declare the variables at first use in most cases and not at the head of the method.